### ISSUE TITLE
Large spike in fan speed on shutdown

### ISSUE THREAD

**Issue by mbway** [tuxedocomputers/tuxedo-control-center#41]
When shutting down, the tuxedo control center daemon closes, and hands control of the fan speed back to the firmware. The fan curve on my laptop's firmware is very aggressive and this causes a massive spike in fan speed while the laptop is shutting down which is very loud.

I think that restoring control to the firmware is the correct behavior when tccd is stopped normally, but when it is stopped because the laptop is shutting down I think it would make sense to give the option to instead transition to a fixed fan speed (like `max(current_speed, 30%)` or something) so long as the current temperature isn't too high.

I wrote a service file to determine whether a script can be run at system shutdown but before daemons are killed:
```
[Unit]
Description=on shutdown
DefaultDependencies=no
Before=shutdown.target reboot.target halt.target

[Service]
Type=oneshot
ExecStart=/bin/bash -c 'ps -ax > /processes'

[Install]
WantedBy=shutdown.target reboot.target halt.target
```

and after rebooting, I searched through `/processes` and found that at the point this script was run, tccd was still running (but so was `tccd --stop`). So it might be possible to have a service run which sets some volatile/temporary configuration instructing tccd what to do when it exits? or perhaps if there is a race condition, the script could instead wait for tccd to close if it hasn't already then quickly set the fan speed to the correct value before the firmware has a chance to alter it?

**Comment by tuxedoxt** [comment]
Yes this is known and a bit unfortunate. Especially noticeable on a few devices with extra loud inherent "hardware" fan control.

As you say it is a tricky topic. The suggestion of a fixed medium speed is interesting. You'd have to be careful only doing it on shutdown, hope no heavy work is left to be done and/or then rely on throttling.

Implementation-wise I would prefer the logic to be a part of tccd, for a shutdown hook. Could be useful in general. Possibly through a separate service file with an alternate stop on shutdown/halt target providing a switch to tccd.

**Comment by latgarf** [comment]
I've also experienced spike in fan speed on suspend (Fn + F12) on Tuxedo Book XP14 - Gen12.  Would be nice if it's solved.

**Comment by MapManagement** [comment]
> Yes this is known and a bit unfortunate. Especially noticeable on a few devices with extra loud inherent "hardware" fan control.
> 
> As you say it is a tricky topic. The suggestion of a fixed medium speed is interesting. You'd have to be careful only doing it on shutdown, hope no heavy work is left to be done and/or then rely on throttling.
> 
> Implementation-wise I would prefer the logic to be a part of tccd, for a shutdown hook. Could be useful in general. Possibly through a separate service file with an alternate stop on shutdown/halt target providing a switch to tccd.

Hey @tuxedoxt ,
is there any fix for now? Since I'm using an Aura15 I had to disable the TCC fan control to prevent the fans from ramping up as soon as the device is shutting down or goes into sleep mode.
I also found some similar issue on the tuxedo-keyboard repository:
https://github.com/tuxedocomputers/tuxedo-keyboard/issues/128
